### PR TITLE
Added renderOptions to render's begin function.

### PR DESCRIPTION
**Pull request by ggarra13**
Created RenderOptions.h, RenderOptionsInline.h and modified IRender.h, tlGL/Render.h and tlGL/Render.cpp to use it in the begin() option.

### CODE REVIEW

File: lib/tlGL/Render.cpp [hunk 1]
```diff
@@ -716,15 +716,19 @@ namespace tl
...
             TLRENDER_P();
 
             p.size = size;
 
-            glViewport(0, 0, p.size.w, p.size.h);
-            glClearColor(0.F, 0.F, 0.F, 0.F);
-            glClear(GL_COLOR_BUFFER_BIT);
+            if ( renderOptions.clear )
+            {
+                glViewport(0, 0, p.size.w, p.size.h);
```
Comment (darbyjohnston) [inline lib/tlGL/Render.cpp:728]: I believe the "glViewport()" call should be left out of this conditional, it sets up the viewport for rendering.

File: lib/tlGL/Render.h [hunk 2]
```diff
@@ -28,7 +28,8 @@ namespace tl
             void setTextureCacheSize(size_t) override;
             void setColorConfig(const timeline::ColorConfigOptions&) override;
             void setLUT(const timeline::LUTOptions&) override;
-            void begin(const imaging::Size&) override;
+            void begin(const imaging::Size&,
```
Comment (darbyjohnston) [inline lib/tlGL/Render.h:31]: Same style comment as above.

File: lib/tlTimeline/IRender.h [hunk 3]
```diff
@@ -41,7 +42,8 @@ namespace tl
             virtual void setLUT(const LUTOptions&) = 0;
 
             //! Start a render.
-            virtual void begin(const imaging::Size&) = 0;
+            virtual void begin(const imaging::Size&,
```
Comment (darbyjohnston) [inline lib/tlTimeline/IRender.h:45]: Same style comment as above.

File: lib/tlTimeline/RenderOptions.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (darbyjohnston) [inline lib/tlTimeline/RenderOptions.h:1]: This reminds me that I need to look into converting the copyright headers to what the ASWF recommends:
https://github.com/AcademySoftwareFoundation/tac/blob/main/process/contributing.md#copyright-notice-format

File: lib/tlTimeline/RenderOptionsInline.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: BSD-3-Clause
```
Comment (darbyjohnston) [inline lib/tlTimeline/RenderOptionsInline.h:1]: The new .h files should be added to the CMakeLists,txt so that CMake knows about them.

**Review by darbyjohnston — CHANGES_REQUESTED** [review]
Hi, thanks for making these changes! After this PR is finished I'll take a look at migrating the OCIO information and other options to use this new struct, I think it will clean up the API quite a bit.

**Comment by ggarra13** [comment]
@darbyjohnston I modified the code as requested.  I also modified the copyright comment in the two new .h files.  I left the other ones as they were.

**Comment by darbyjohnston** [comment]
Sorry, please do not use the ASWF headers yet, that was just a note that I need to look into it further. I'll make a global change later to change all of the headers at once.

**Comment by ggarra13** [comment]
@darbyjohnston Fixed it again.

**Comment by darbyjohnston** [comment]
Thanks again!

**Pull request merged**